src/unix/internal.h

```diff
@@ -125,9 +125,35 @@ typedef struct uv__stream_queued_fds_s uv__stream_queued_fds_t;
 
 /* loop flags */
 enum {
-  UV_LOOP_BLOCK_SIGPROF = 1
+  UV_LOOP_BLOCK_SIGPROF = 1,
+  /* data member of backend union is a uv__backend_data_io_uring */
+  UV_LOOP_BACKEND_DATA_IO_URING = (1 << 1)
```

Comment:
`(1u << 31)` isn't allowed as an enum value by ISO C. Rather it be `(1u << 30)` or `(int)(1u << 31)`?

I'm confused what you mean in the second part; I don't understand how the runtime branches can be replaced by a compile-time `#if` given that the flag is set during loop init. What do you mean?

Reply:
I guess that's true in pedantic C mode (it isn't in C++) so `1 << 30` then.

About my "remove the branches" comment, I should perhaps have added "on platforms other than linux" :-) - the compiler can remove the branch because `flags & 0` is always zero.